dhall/src/Dhall/Main.hs

```diff
@@ -295,14 +305,54 @@ parseMode =
     parseInplace = fmap f (optional p)
       where
         f  Nothing    = StandardInput
-        f (Just file) = File file
+        f (Just file) = InputFile file
 
         p = Options.Applicative.strOption
             (   Options.Applicative.long "inplace"
             <>  Options.Applicative.help "Modify the specified file in-place"
             <>  Options.Applicative.metavar "FILE"
             )
 
+    parseInput = fmap f (optional p)
+      where
+        f  Nothing    = StandardInput
+        f (Just path) = InputFile path
+
+        p = Options.Applicative.strOption
+            (   Options.Applicative.long "path"
+            <>  Options.Applicative.help "Index all files in path recursively. Will get list of files from STDIN if ommited."
+            <>  Options.Applicative.metavar "INPUT"
```

Comment:
@sjakobi , why? path can be path to file or directory or `StandardInput`.
In case of file, we will have tags for file, in case directory, `generate` will index whole directory and for `StandardInput` it will read list of files from `stdin`.

Reply:
Ok. How about `PATH` then? You can't specify `stdin` with `--path`, right?

Reply:
No, it was before and was changed.
Now to use `stdin` you can omit this parameter, in other case it will be path.
I think `PATH` is good, I'll change.

Reply:
Done